app/models/clearance_batch.rb

```diff
@@ -5,33 +5,15 @@ class ClearanceBatch < ActiveRecord::Base
   has_many :items
   attr_accessor :item_errors
 
-  def clearance_items!(potential_item_ids)
-    self.item_errors = []
-    if potential_item_ids.any?
-      Item.transaction do
-        potential_item_ids.each do |potential_item_id|
-          if item_id_is_valid?(potential_item_id)
-            item = Item.find(potential_item_id)
-            item.clearance!
-            self.items << item
-          end
-        end
-      end
-    end
-  end
-
   def item_id_is_valid?(potential_item_id)
```

Comment:
I like the changes that make this more functional. What do you think about moving this method into the `ClearancingService` class as well to concentrate the business logic there? That could also cut down on the length of the lines where you're calling it there, making it a bit easier to read.